R/classify.R

```diff
@@ -0,0 +1,57 @@
+# Function to calculate mode
+calculate_mode <- function(x) {
+  ux <- unique(x)
+  ux[which.max(tabulate(match(x, ux)))]
+}
+
+# Function to calculate mode behavior for each window of N minutes
+plot_mode_behavior_by_window <- function(data,
+                                         time_col,
+                                         behavior_col,
+                                         window_minutes) {
+  data %>%
+    dplyr::mutate(time = ymd_hms(!!sym(time_col)),
+                  # Ensure the time column is in POSIXct format
+                  window = floor_date(time, unit = paste0(window_minutes, " mins"))) %>%
+    dplyr::group_by(window) %>%
+    dplyr::summarise(mode_behavior = calculate_mode(!!sym(behavior_col))) %>%
+    dplyr::filter(!is.na(mode_behavior)) -> mode_sum
+  mode_sum %>%
+    ggplot2::ggplot(ggplot2::aes(x = window, y = mode_behavior, color = mode_behavior)) +
+    ggplot2::geom_point(size = 3) +
+    ggplot2::labs(
+      x = "Time",
+      y = "Mode Behavior",
+      title = paste0("Mode Behavior Every ", window_minutes, " Minutes")
+    ) +
+    ggplot2::theme_minimal() -> plotting
+  print(plotting)
+  return(mode_sum)
```

Comment:
For the return, I wonder if if should return both data and the plot? What if you want to edit the plot? 

Could also be invisible?

Reply:
should probably be two functions to do it properly?

Reply:
plotting is so trivial, pulling out for now